### PR TITLE
Additional error handling for cw es / data value resolvers

### PR DESCRIPTION
**Pull request by cragun47**
This satisfies some additional edge conditions related to CodeableConcepts, CWE's, and graceful fallback.  (#173)

### CODE REVIEW

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 1]
```diff
@@ -326,17 +331,28 @@ public void patient_communication_language(){
...
-    Patient patientObjNoCode = PatientUtils.createPatientFromHl7Segment(patientEnglishNoCode);
-    assertThat(patientObjNoCode.hasCommunication()).isTrue();
-    assertThat(patientObjNoCode.getCommunication().get(0).getPreferred()).isTrue();
-    assertThat(patientObjNoCode.getCommunication()).hasSize(1);
-    Patient.PatientCommunicationComponent ccNoCode = patientObjNoCode.getCommunication().get(0);
+    Patient patientObjNoSystem = PatientUtils.createPatientFromHl7Segment(patientEnglishNoSystem);
+    assertThat(patientObjNoSystem.hasCommunication()).isTrue();
+    assertThat(patientObjNoSystem.getCommunication().get(0).getPreferred()).isTrue();
+    assertThat(patientObjNoSystem.getCommunication()).hasSize(1);
+    Patient.PatientCommunicationComponent ccNoCode = patientObjNoSystem.getCommunication().get(0);
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:338]: Add check for display

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 2]
```diff
@@ -326,17 +331,28 @@ public void patient_communication_language(){
...
+    Patient patientObjCodeOnly = PatientUtils.createPatientFromHl7Segment(patientEnglishCodeOnly);
+    assertThat(patientObjCodeOnly.hasCommunication()).isTrue();
+    assertThat(patientObjCodeOnly.getCommunication().get(0).getPreferred()).isTrue();
+    assertThat(patientObjCodeOnly.getCommunication()).hasSize(1);
+    Patient.PatientCommunicationComponent ccCodeOnly = patientObjCodeOnly.getCommunication().get(0);
+    assertThat(ccCodeOnly.getPreferred()).isTrue();
+    assertThat(ccCodeOnly.getLanguage().hasText()).isFalse();
+    Coding coding = ccCodeOnly.getLanguage().getCodingFirstRep();
+    assertThat(coding.getCode()).isEqualTo("ENG");
+    assertThat(coding.getSystem()).isNull();
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:354]: Add check for display

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 3]
```diff
@@ -326,17 +331,28 @@ public void patient_communication_language(){
     assertThat(code.getCode()).isEqualTo("ENG");
     assertThat(code.getSystem()).isEqualTo("urn:id:v2-0296");
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:332]: Add check for display

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 4]
```diff
@@ -281,41 +281,46 @@ public void patient_gender_test() {
...
+    Patient patientObjMarried = PatientUtils.createPatientFromHl7Segment(marriedPatientWithVersion);
     assertThat(patientObjMarried.hasMaritalStatus()).isTrue();
     assertThat(patientObjMarried.getMaritalStatus().getCodingFirstRep().getDisplay()).isEqualTo(V3MaritalStatus.M.getDisplay());
     assertThat(patientObjMarried.getMaritalStatus().getCodingFirstRep().getSystem()).isEqualTo(V3MaritalStatus.M.getSystem());
+    assertThat(patientObjMarried.getMaritalStatus().getCodingFirstRep().getVersion()).isEqualTo("47");
 
-    Patient patientObjMarriedAltText = PatientUtils.createPatientFromHl7Segment(AltTextField);
+    Patient patientObjMarriedAltText = PatientUtils.createPatientFromHl7Segment(singlePatientWithVersion);
     assertThat(patientObjMarriedAltText.hasMaritalStatus()).isTrue();
     assertThat(patientObjMarriedAltText.getMaritalStatus().getText()).isEqualTo("Never Married");
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:303]: Add checks for system, display

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java [hunk 5]
```diff
@@ -281,41 +281,46 @@ public void patient_gender_test() {
...
             "MSH|^~\\&|MyEMR|DE-000001| |CAIRLO|20160701123030-0700||VXU^V04^VXU_V04|CA0001|P|2.6|||ER|AL|||||Z22^CDCPHINVS|DE-000001\r" +
-                    "PID|1||000054321^^^MRN||COOPER^SHELDON^ANDREW||19820512|M^Married||2106-3|765 SOMESTREET RD UNIT 3A^^PASADENA^LA^558846^United States of America||4652141486^Home^^shelly@gmail.com||EN^English|S|CAT|78654||||N\r";
+                    "PID|1||12345678^^^MRN||TestPatient^Jane|||||||||||S^^^^^^1.1||||||\r";
 
-    Patient patientObjMarried = PatientUtils.createPatientFromHl7Segment(marriedPatient);
+    Patient patientObjMarried = PatientUtils.createPatientFromHl7Segment(marriedPatientWithVersion);
     assertThat(patientObjMarried.hasMaritalStatus()).isTrue();
     assertThat(patientObjMarried.getMaritalStatus().getCodingFirstRep().getDisplay()).isEqualTo(V3MaritalStatus.M.getDisplay());
     assertThat(patientObjMarried.getMaritalStatus().getCodingFirstRep().getSystem()).isEqualTo(V3MaritalStatus.M.getSystem());
+    assertThat(patientObjMarried.getMaritalStatus().getCodingFirstRep().getVersion()).isEqualTo("47");
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:299]: Add check for text

**Review by LisaWellman — APPROVED** [review]
Looks good, just a couple of suggestions for extra validation in some tests

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 1 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:338]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 3 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:332]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 4 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:303]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 5 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:299]: Done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java (hunk 2 above)
Comment (cragun47) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7PatientFHIRConversionTest.java:354]: Done
